apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -45,6 +64,103 @@ export const jsonFormsLinkControlTester: RankedTester = rankWith(
   ),
 )
 
+interface PageLinkModalProps {
+  data: string
+  description?: string
+  isOpen: boolean
+  onClose: () => void
+  onSave: (destination: string) => void
+}
+
+const PageLinkModal = ({
+  data,
+  description,
+  isOpen,
+  onClose,
+  onSave,
+}: PageLinkModalProps) => {
+  const { handleSubmit, setValue, watch } = useZodForm({
+    schema: z.object({
+      destination: z.string(),
+    }),
+    defaultValues: {
+      destination: "",
+    },
+  })
+
+  const onSubmit = handleSubmit(({ destination }) => onSave(destination))
+  const destination = watch("destination")
+
+  const { data: selectedResource } =
+    trpc.resource.getWithFullPermalink.useQuery({
+      resourceId: destination,
+    })
+
+  useEffect(() => {
+    // set default values here instead
+    setValue("destination", getResourceIdFromReferenceLink(data))
+    // only done once per every time the modal is opened
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen])
```

Comment:
a way to mitigate this is to move all the form logic into a separate `PageLinkModalContent` component, then rerender that component whenever the modal is reopened. Then the default values will always be reset.

```tsx
<Modal>
  {isOpen && <PageLinkModalContent {...} />
</Modal>
```

See [Rerender form instead of using useEffect to reset values](https://www.notion.so/opengov/Rerender-form-instead-of-using-useEffect-to-reset-values-430eb644c068469e8cae68c4988dfc31)

Reply:
Updated in 1a92e6aab9d260070adc42fafda564b266228d9f.